src/native-state-drm.cpp

```diff
@@ -195,8 +198,100 @@ NativeStateDRM::init_gbm()
     return true;
 }
 
-bool
-NativeStateDRM::init()
+char const *
+NativeStateDRM::drm_primary_gpu_device_node
+(struct udev * __restrict const udev,
+ struct udev_enumerate * __restrict const dev_enum)
+{
+    char const * __restrict result = NULL;
+
+    struct udev_list_entry * device_list =
+        udev_enumerate_get_list_entry(dev_enum);
+
+    struct udev_list_entry * __restrict current_element;
+
+    udev_list_entry_foreach(current_element, device_list) {
+        
+        char const * __restrict current_element_sys_path =
+            udev_list_entry_get_name(current_element);
+        
+        struct udev_device * __restrict const current_device =
+            udev_device_new_from_syspath(udev, current_element_sys_path);
+        
+        struct udev_device * __restrict const potential_primary_gpu =
+            udev_device_get_parent_with_subsystem_devtype(
+                current_device, "pci", NULL
+            );
```

Comment:
Here and elsewhere, maintain the formatting style of the file:

Either:
```
udev_device_get_parent_with_subsystem_devtype(current_device, "pci", NULL);
```
or:
```
udev_device_get_parent_with_subsystem_devtype(current_device,
                                              "pci",
                                              NULL);
```
Or if the line is overly long:

```
udev_device_get_parent_with_subsystem_devtype(
    current_device, "pci", NULL);
```